src/pybind/mgr/volumes/fs/stats_util.py

```diff
@@ -0,0 +1,188 @@
+                       'but this is not sufficient in this case. '
+                      f'Current size = {num}')
+
+        x, y = str(num).split('.')
+        if len(x) <= 3:
+            if y:
+                y = '' if y == '0' else y[:4]
+
+            return f'{x}.{y} {size[i]}' if y else f'{x} {size[i]}'
+
+
+def get_num_h(num):
+    '''
+    Convert number to human-readable format.
+    '''
+    size = ('', 'K', 'M', 'B', 'T')
+
+    if isinstance(num, bytes):
+        num = int(num)
+
+    if isinstance(num, str):
+        if not num.isdigit():
+            return num
+        num = int(num)
+
+    if num < 1000:
+        return f'{num}'
+
+    i = 0
+    while True:
+        num = num / 1000
+        i += 1
+        if i >= len(size):
+            log.error(f"Biggest suffix we have is '{len(size)-1}' "
+                       'but this is not sufficient in this case. '
+                      f'Current size = {num}')
+
+        x, y = str(num).split('.')
+        if len(x) <= 3:
+            if y:
+                y = '' if y == '0' else y[:4]
+
+            return f'{x}.{y} {size[i]}' if y else f'{x} {size[i]}'
+
+
+def get_size_ratio_str(size1, size2, human=True):
+    if human:
+        size1, size2 = get_size_h(size1), get_size_h(size2)
+
+    return f'{size1}/{size2}'
+
+
+def get_num_ratio_str(num1, num2, human=True):
+    if human:
+        num1, num2 = get_num_h(num1), get_num_h(num2)
+
+    return f'{num1}/{num2}'
+
+
+def get_amount_copied(src_path, dst_path, fsh, human=True):
+    rbytes = 'ceph.dir.rbytes'
+
+    size_t = int(fsh.getxattr(src_path, rbytes))
+    size_c = int(fsh.getxattr(dst_path, rbytes))
+
+    if size_t == 0:
+        return -1 if size_c == 0 else -2
+
+    percent = ((size_c/size_t) * 100)
+    percent = round(percent, 3)
+
+    return size_t, size_c, percent
+
+
+def get_stats(src_path, dst_path, fsh, human=True):
+    rfiles = 'ceph.dir.rfiles'
+    rsubdirs = 'ceph.dir.rsubdirs'
+
+    rfiles_t = int(fsh.getxattr(src_path, rfiles))
+    subdirs_t = int(fsh.getxattr(src_path, rsubdirs))
+
+    rfiles_c = int(fsh.getxattr(dst_path, rfiles))
+    subdirs_c = int(fsh.getxattr(dst_path,rsubdirs))
+
+    size_t, size_c, percent = get_amount_copied(src_path, dst_path, fsh,
+                                                human)
+
+    return {
+        'percentage cloned': percent,
+        'amount cloned': get_size_ratio_str(size_c, size_t, human),
+        'regfiles cloned': get_num_ratio_str(rfiles_c, rfiles_t, human),
+        'subdirs cloned': get_num_ratio_str(subdirs_c, subdirs_t, human)
+    }
+
+
+class CloneProgressBar:
+
+    def __init__(self, vc, volname, dst_svname, src_path, dst_path):
+        self.vc = vc
+        self.volname = volname
+        self.dst_svname = dst_svname
+
+        self.src_path = src_path
+        self.dst_path = dst_path
+
+        self.pev_id = str(uuid4())
+
+        # FIXME: when function name is incorrectly written, the error
```

Comment:
> The undefined function exception was caught and logged.

That's not the concern here. The concern is that neither a traceback is printed in the logs nor a crash occurs. This makes is very very easy to miss it. Plus, only error message of exception is printed which makes debugging really unnecessarily difficult since it becomes vague and ambiguous.

Reply:
If you want the trackback to be printed then you could modify RTimer to log the whole traceback and the exception. It's off-putting to explain that as part of a comment esp. when the issue was an incorrect function name that was used during testing.